kingpin/constants.py

```diff
@@ -46,3 +48,18 @@ class STATE(StringCompareBase):
     """
 
     valid = ('present', 'absent')
+
+
+class SchemaCompareBase(object):
```

Comment:
I refactored the code a bit .. it can be continually improved, but lets focus on that outside of this PR.